builder/comp-builder.nix

```diff
@@ -220,6 +224,7 @@ let
       inherit configFiles executableToolDepends cleanSrc exeName;
       env = shellWrappers;
       profiled = self (drvArgs // { enableLibraryProfiling = true; });
```

Comment:
Your observation about the problem with `covered` makes me wonder if *this* works. What happens if I do `someTestComponent.profiled`? Do I get the test built with profiling, but not the library? :thinking:

Reply:
Yeah, so that's a really good question.

I've figured out the issue I was encountering with `covered` so I understand this a bit more. The key step I was missing was this line in the `comp-builder.nix` file: https://github.com/input-output-hk/haskell.nix/blob/e340658306cea5817f5478ad4ad8ba4f7da2381f/builder/comp-builder.nix#L76

So when you build `someTestComponent.profiled`, the component builder enables library profiling on that derivation. That's enough to profile the test (and maybe it's immediate dependencies? I don't know how "--enable-library-profiling" works). But it's important that the request for profiling is then fed to `make-config-files.nix`, which will then call the `profiled` attribute on the dependencies of the test component, and this will recurse down the entire dependency tree: https://github.com/input-output-hk/haskell.nix/blob/e340658306cea5817f5478ad4ad8ba4f7da2381f/builder/make-config-files.nix#L50

I needed to do something very similar with the `covered` attribute, but I only wanted it to apply to the immediate library dependency of the test component. The user doesn't care if their test fully covers the `bytestring` or `transformers` library. I've got this working here: https://github.com/input-output-hk/haskell.nix/pull/762/commits/2087cbf0c8b72447f65303a846de09f6f1e12b0e, but it's quite inelegant.

cc @michaelpj

Reply:
Yikes! We should really have a big comment explaining how this works, sigh.

I'm not a huge fan of this approach but I don't have a better suggestion at the moment.